docs/reference/module-api/module-overview.md

```diff
@@ -908,12 +826,146 @@ module.exports = {
+            .toArray();
+
+          return result[0];
+        }
+      }
+    };
+  }
+};
+```
+
 #### `extendQueries(self)`
 
+Extend the behavior of existing event handlers (set in `queries`) in the `extendQueries` section. This function must return an object as described in `queries`.
+
+Each extended query builder or method should accept the original function as `_super` followed by its original arguments, if there are any. Extended query builders and methods will be matched with the base class builder or method using the same name. Methods should return data in a similar format to the existing API route.
+
 ### `middleware(self)`
 
+Add standard Express middleware to be called on *every* request. The `middleware` function takes the module as an argument and must return an object of [middleware functions](https://expressjs.com/en/guide/using-middleware.html). This is a good place to import third-party middleware into Apostrophe.
+
+Note that it can often be simpler to add an event handler or `await` a method in an API route instead.
+
+`ours` could also be an object with a `before` property and a `middleware` property, in which case it would run `before` the middleware of the module with the specified name
+
+```javascript
+// modules/limiter/index.js
+module.exports = {
+  // ...
+  middleware(self, options) {
+    return {
+      ours(req, res, next) {
+        // Restrict access by IP address, in a crude way.
+        const allowlist = [ '127.0.0.1', '::1' ];
+
+        if (!allowlist.includes(req.connection.remoteAddress)) {
+          return res.status(403).send('forbidden');
+        }
+        return next();
+      }
+    };
+  }
+};
+```
+
+If the middleware function must run before another named middleware function, set the returned object key to an object with `before` and `middleware` properties. `before` would be set to the name of a module whose middleware must run after the new function, which is now the value of `middleware`.
+
+```javascript
+// modules/limiter/index.js
+module.exports = {
+  // ...
+  middleware(self, options) {
+    return {
+      ours: {
+        // 👇 Same as above, but with `before` and `middleware` properties.
+        before: '@apostrophecms/login',
+        middleware: function (req, res, next) {
+          // Restrict access by IP address, in a crude way.
+          const allowlist = [ '127.0.0.1', '::1' ];
+
+          if (!allowlist.includes(req.connection.remoteAddress)) {
+            return res.status(403).send('forbidden');
+          }
+          return next();
+        }
+      }
+    };
+  }
+};
+```
+
 ### `tasks(self)`
 
-## "Cascading" settings
+`tasks` takes the module as an argument and returns an object of functions that define command-line tasks. Task properties include:
+
+
+| Property | Description |
+| ------- | ------- |
+| `usage` | A string describing the task and how to use it. It is printed on the command-line. |
+| `task` | The task function. It maybe asynchronous. |
```

Comment:
```suggestion
| `task` | The task function. Can be asynchronous. |
```
or 

```suggestion
| `task` | The task function. It may be asynchronous. |
```